src/main/java/io/vertx/openapi/validation/impl/RequestValidatorImpl.java

```diff
@@ -153,6 +154,28 @@ RequestParameter validateBody(RequestBody requestBody, ValidatableRequest reques
       throw new ValidatorException("The format of the request body is not supported", UNSUPPORTED_VALUE_FORMAT);
     }
     Object transformedValue = transformer.transformRequest(mediaType, request);
+    // Corner case for "application/octet-stream" media type:
+    // Skip full schema validation if the following simplified validation succeeds,
+    // otherwise validate with JSON Validator and fail the validation
+    // of the transformed value eventually.
+    if (mediaType.getIdentifier().equals(MediaType.APPLICATION_OCTET_STREAM)) {
+      JsonSchema schema = mediaType.getSchema();
+      String schemaTypeValue = schema.get("type", null);
+      if (schemaTypeValue != null && schemaTypeValue.equals("string")) {
+        String schemaFormatValue = schema.get("format", null);
+        // Case for OpenAPI 3.0 and OpenAPI 3.1 (backwards compatibility)
+        if (schemaFormatValue != null && schemaFormatValue.equals("binary")) {
+          return new RequestParameterImpl(transformedValue);
+        } else {
+          // Case only for OpenAPI 3.1
+          String schemaContentMediaType = schema.get("contentMediaType", null);
+          if (schemaContentMediaType != null
+            && schemaContentMediaType.equals(MediaType.APPLICATION_OCTET_STREAM)) {
+            return new RequestParameterImpl(transformedValue);
```

Comment:
I don't agree to this. As long as the type is `string`, you could have properties like `maxLength`, couldn't you?

Reply:
Sorry I don't understand that one?

The `null` check for `conentMediaType` is because of the support for OpenAPI 3.1. Someone could use a contract file for version 3.0 where that field is not a requirement.

The verification of the value of `contentMediaType` is necessary because it could have another value.

Reply:
What I mean is, that you skip the validation. Couldn't it be possible to define a schema like this?

```
type: string
contentMediaType: application/octet-stream
minLength: 2
```